### DISCUSSION TITLE
How do you nest ordered lists with different symbols

### DISCUSSION THREAD

**Discussion by web-shenjin** [Milkdown/milkdown#698] [Q&A]
[
![微信图片_20220830202746](https://user-images.githubusercontent.com/52124289/187441356-7ea94a2b-6b56-4b82-a5bd-3221a5af965e.jpg)
](url)
How do you nest ordered lists with different symbols
eg 
1. apple0-1
2. apple0-2
3. apple0-3
   a.apple-1-1
   b.apple-1-2
   c.apple-1-3
      i.apple2-1
    ii,apple2-2
   iii.apple2-3
       1.apple3-1
       2.apple3-2
      3.apple3-3

**Comment by z-950** [comment]
Create your own `list-item` to replace the `preset-commonmark`'s by do something at [here](https://github.com/Saul-Mirone/milkdown/blob/d568bf0784e33af38ba88c007bd9d58b5f598a28/packages/preset-commonmark/src/node/list-item.ts#L37-L45)

**Reply by web-shenjin** [reply]
![Snipaste_2022-09-05_13-59-17](https://user-images.githubusercontent.com/52124289/188370719-c0362de0-ebfd-4463-bab5-5122869463f2.png)

this is params utils is doing what?

**Reply by web-shenjin** [reply]
How to implement a example plugins
eg: 
a.xxx
b:xxx1
c:xxx2

**Reply by web-shenjin** [reply]
i write own plugins, utils this params is not understand

**Reply by web-shenjin** [reply]
> Create your own `list-item` to replace the `preset-commonmark`'s by do something at [here](https://github.com/Saul-Mirone/milkdown/blob/d568bf0784e33af38ba88c007bd9d58b5f598a28/packages/preset-commonmark/src/node/list-item.ts#L37-L45)

How to implement a example plugins
eg:
a.xxx
b.xxx1
c.xxx2

**Comment by web-shenjin** [comment]
How to implement a example plugins
eg:
  a.xxx
  b.xxx1
  c.xxx2

**Comment by z-950** [comment]
Like this:
```js
attrs['label'] = String.fromCharCode(97+index) + '.'
```

**Reply by web-shenjin** [reply]
使用你说的，页面会卡死。
Infinite loop, stuck

**Reply by web-shenjin** [reply]
![Snipaste_2022-09-05_16-52-39](https://user-images.githubusercontent.com/52124289/188410040-56589662-8d8b-437a-ad60-cdbe4a6769d0.png)
![Snipaste_2022-09-05_16-53-37](https://user-images.githubusercontent.com/52124289/188410052-03dc6049-0654-4f42-90f2-fcf7ce9c7306.png)
![Snipaste_2022-09-05_16-53-47](https://user-images.githubusercontent.com/52124289/188410054-257018ac-56c2-4e7c-9da9-59cc40584867.png)

**Reply by z-950** [reply]
![image](https://user-images.githubusercontent.com/30150653/188528480-c76d30d9-cf9a-4ff4-b610-2c9a63f91728.png)
This code works for me. You should check some other places.

**Reply by web-shenjin** [reply]
please give me, you finish vue version example？ OK? thank

**Reply by z-950** [reply]
You should not import the official list-item.

**Reply by web-shenjin** [reply]
you write vue3 example

**Reply by web-shenjin** [reply]
How did you implement it? Can you provide the code

**Comment by web-shenjin** [comment]
How did you implement it? Can you provide the code

**Comment by z-950** [comment]
> Create your own `list-item` to replace the `preset-commonmark`'s by do something at [here](https://github.com/Saul-Mirone/milkdown/blob/d568bf0784e33af38ba88c007bd9d58b5f598a28/packages/preset-commonmark/src/node/list-item.ts#L37-L45)

It's only about milkdown.

Read `preset-commonmark` code and you will get it.

```js
import {
    InsertHardbreak,
    InsertHr,
    InsertImage,
    LiftListItem, // This should be your custom list-item
    ModifyImage,
    nodes,
    SinkListItem,
    SplitListItem,
    TurnIntoCodeFence,
    TurnIntoHeading,
    TurnIntoText,
    WrapInBlockquote,
    WrapInBulletList,
    WrapInOrderedList,
} from './node';
import { commonmarkPlugins } from './plugin';

// ...

export const commonmarkNodes = AtomList.create([...nodes, ...marks]);
export { commonmarkPlugins };
export const commonmark = AtomList.create([...commonmarkPlugins, ...commonmarkNodes]);
```
